### PR TITLE
Change workflow to mamba

### LINKED ISSUE #397: Change workflow from `pip` to `mamba/boa`
Change workflow from `pip` to `mamba/boa`

### PR DESCRIPTION
**Pull request by forsyth2**
Change workflow to mamba. Resolves #397.

### CODE REVIEW

**Comment by forsyth2** [comment]
@chengzhuzhang Once this pull request and #428 merged, I can probably create equivalent pull requests for both `zstash` and `E3SM Diags`.

@tomvothecoder I believe in our meeting, we discussed that the only necessary change was to remove the `intall_requires=["configobj>=5.0.0,<6.0.0", "jinja2>=2.0.0"],` line. I have removed that here.

`git grep -n pip` shows a number of matches. So we're ok with keeping these instances?
```
.github/workflows/build_workflow.yml:81:        run: pip install .
.github/workflows/build_workflow.yml:106:      - name: Cache pip
.github/workflows/build_workflow.yml:110:          path: ~/.cache/pip
.github/workflows/build_workflow.yml:112:          key: ${{ runner.os }}-pip-publish-docs
.github/workflows/build_workflow.yml:114:            ${{ runner.os }}-pip-
.github/workflows/build_workflow.yml:117:      # Using pip for Sphinx dependencies because it takes too long to reproduce a conda environment (~10 secs vs. 3-4 mins)
.github/workflows/build_workflow.yml:120:          python -m pip install --upgrade pip
.github/workflows/build_workflow.yml:121:          pip install sphinx==5.2.3 sphinx_rtd_theme==1.0.0 docutils==0.16
.github/workflows/release_workflow.yml:26:      - name: Cache pip
.github/workflows/release_workflow.yml:30:          path: ~/.cache/pip
.github/workflows/release_workflow.yml:32:          key: ${{ runner.os }}-pip-publish-docs
.github/workflows/release_workflow.yml:34:            ${{ runner.os }}-pip-
.github/workflows/release_workflow.yml:37:      # Using pip for Sphinx dependencies because it takes too long to reproduce a conda environment (~10 secs vs. 3-4 mins)
.github/workflows/release_workflow.yml:40:          python -m pip install --upgrade pip
.github/workflows/release_workflow.yml:41:          pip install sphinx==5.2.3 sphinx_rtd_theme==1.0.0 docutils==0.16
conda/dev.yml:9:  - pip=22.2.2
conda/dev.yml:33:  - pip:
conda/meta.yaml:14:  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
conda/meta.yaml:22:    - pip
conda/meta.yaml:32:    - pip
conda/meta.yaml:37:    - pip check
docs/source/dev_guide/new_glb_plot.rst:13:- Run the `integration tests <https://e3sm-project.github.io/zppy/_build/html/main/dev_guide/testing.html#integration-tests>`_ and examine the differences from the expected files. If they match what you expect, update the expected files following "Commands to run to replace outdated expected files" on the machine-specific directions. (The commands to run all the integration tests are ``pip install .`` followed by ``python -u -m unittest tests/integration/test_*.py``).
docs/source/dev_guide/testing.rst:12:        pip install . # Install your changes (`python -m pip install .` also works)
docs/source/dev_guide/testing.rst:25:        pip install . # Install your changes (`python -m pip install .` also works)
docs/source/dev_guide/testing_e3sm_unified.rst:130:This will require running ``pip install .`` in that environment from the top level of
docs/source/getting_started.rst:162:Instead, the developer will ``python -m pip install .`` to build ``zppy`` with changes
docs/source/getting_started.rst:248:        pip install .
```

@xylar From #428:
> You are right that the yaml file should also be updated to switch sphinx-multiversion to come from conda-forge, not pip. That seems like a trivial change.

I added a change to use `mamba` to install `sphinx-multiversion` but I'm not sure I did so correctly.

**Comment by xylar** [comment]
@forsyth2, my comment was referring to this yaml file:
https://github.com/E3SM-Project/zppy/blob/main/conda/dev.yml#L33-L34

Here, you will want to install all the packages currently being installed with pip using `mamba` instead, not just `sphinx-multiversion`.  You want to avoid mixing conda-forge and pip packages whenever possible.

File: .github/workflows/build_workflow.yml [hunk 1]
```diff
@@ -118,7 +118,8 @@ jobs:
       - name: Install Dependencies
         run: |
           python -m pip install --upgrade pip
```
Comment (xylar) [inline .github/workflows/build_workflow.yml:120]:
```suggestion
          mamba install -y pip phinx==5.2.3 sphinx_rtd_theme==1.0.0 sphinx-multiversion==0.2.4 docutils==0.16
```

File: .github/workflows/build_workflow.yml [hunk 2]
```diff
@@ -118,7 +118,8 @@ jobs:
...
-          pip install sphinx==5.2.3 sphinx_rtd_theme==1.0.0 sphinx-multiversion==0.2.4 docutils==0.16
+          pip install sphinx==5.2.3 sphinx_rtd_theme==1.0.0 docutils==0.16
+	  mamba install sphinx-multiversion
```
Comment (xylar) [inline .github/workflows/build_workflow.yml:122]:
```suggestion
```

File: .github/workflows/release_workflow.yml [hunk 3]
```diff
@@ -38,7 +38,8 @@ jobs:
       - name: Install Dependencies
         run: |
           python -m pip install --upgrade pip
```
Comment (xylar) [inline .github/workflows/release_workflow.yml:40]:
```suggestion
```

File: .github/workflows/release_workflow.yml [hunk 4]
```diff
@@ -38,7 +38,8 @@ jobs:
...
-          pip install sphinx==5.2.3 sphinx_rtd_theme==1.0.0 sphinx-multiversion==0.2.4 docutils==0.16
+          pip install sphinx==5.2.3 sphinx_rtd_theme==1.0.0 docutils==0.16
+	  mamba install sphinx-multiversion
```
Comment (xylar) [inline .github/workflows/release_workflow.yml:42]:
```suggestion
          mamba install -y pip phinx==5.2.3 sphinx_rtd_theme==1.0.0 sphinx-multiversion==0.2.4 docutils==0.16
```

**Comment by xylar** [comment]
@forsyth2, here is how I am using mambaforge instead of miniconda in my GitHub Actions:
https://github.com/E3SM-Project/polaris/blob/main/.github/workflows/build_workflow.yml#L82

I'd suggest doing that here, too.

**Comment by forsyth2** [comment]
@tomvothecoder @xylar I think this pull request should be good to go. Please let me know if I'm missing anything. Thanks!

**Review by xylar — APPROVED** [review]
@forsyth2, sorry for the delay.  I didn't realize this was waiting on me.  Looks good to me.

**Review by tomvothecoder — APPROVED** [review]
LGTM

**Pull request merged**